### ISSUE TITLE
Implement base model session 

### ISSUE THREAD

**Issue by CCThompson82** [CCThompson82/nuclei#18]
## Objectives 

-  utility to load image and mask together 
-  queue runners for data input 
-  logic for training epochs, running validation, finishing experiment on test set generation

**Comment by CCThompson82** [comment]
[using queues blog](https://blog.metaflow.fr/tensorflow-how-to-optimise-your-input-pipeline-with-queues-and-multi-threading-e7c3874157e0)
